_protected/app/system/core/assets/cron/96h/BannedCoreCron.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+/**
+ * @title            Birthday Cron Class
+ *
+ * @author           Pierre-Henry Soria <ph7software@gmail.com>, Polyna-Maude R.-Summerside <polynamaude@gmail.com>
+ * @copyright        (c) 2013-2018, Pierre-Henry Soria. All Rights Reserved.
+ * @license          GNU General Public License; See PH7.LICENSE.txt and PH7.COPYRIGHT.txt in the root directory.
+ * @package          PH7 / App / System / Core / Asset / Cron / 24H
+ * @version          1.0
+ */
+namespace PH7;
+
+defined ( 'PH7' ) or exit ( 'Restricted access' );
+
+use Exception;
+use PH7\Framework\Security\Ban\Ban;
+use PH7\Framework\Error\Logger;
+
+class BannedCoreCron extends Cron {
+	/**
+	 * Web client used to fetch IPs
+	 *
+	 * @var \GuzzleHttp\Client
+	 */
+	private $webClient;
+
+	/**
+	 * Banned IP filename
+	 *
+	 * @var string
+	 */
+	private $bannedFileName;
+	/**
+	 * Contain new blocked IP just fetched
+	 *
+	 * @var array
+	 */
+	private $ipNew;
+
+	/**
+	 * Contain existing blocked IP
+	 *
+	 * @var array
+	 */
+	private $ipOld;
+
+	/**
+	 * IP extracting regulat expression
+	 *
+	 * @var string
+	 */
+	private $ipRegExp;
+
+	/**
+	 * Contain the URL of the service we call to get banned IP
+	 * Currently filled at instantiation staticly, will use config file later
+	 *
+	 * @var array
+	 */
+	private $svcUrl;
+	public function __construct() {
+		parent::__construct ();
+
+		/**
+		 * Set valid IP regular expression using lazy mode (false)
+		 *
+		 * @var \PH7\BannedCoreCron $ipRegExp
+		 */
+		$this->ipRegExp = self::regexpIP ();
+
+		/**
+		 * Set the banned IP file name to the same file as used by PH7\Framework\Security\Ban\Ban
+		 *
+		 * @var \PH7\BannedCoreCron $bannedFileName
+		 */
+		$this->bannedFileName = PH7_PATH_APP_CONFIG . Ban::DIR . Ban::IP_FILE;
+		/**
+		 *
+		 * @var \PH7\BannedCoreCron $svcUrl
+		 */
+		$this->svcUrl = array (
```

Comment:
You could store those two URLs into a constant array (since pH7CMS supports 5.6+) like below
```php
const SVC_URLS = [
    'https://www.blocklist.de/downloads/export-ips_all.txt',
    'http://www.badips.com/get/list/ssh/2'
];

it will be clearer and more readable if other devs want to add more urls to the list.